framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -73,9 +75,10 @@ public abstract class AnnotatedTypeMirror {
   // AnnotationMirror doesn't override Object.hashCode, .equals, so we use
   // the class name of Annotation instead.
   // Caution: Assumes that a type can have at most one AnnotationMirror for any Annotation type.
-  protected final AnnotationMirrorSet annotations = new AnnotationMirrorSet();
+  // Not final so clone() can assign it; no other method should assign it.
+  protected /*final*/ AnnotationMirrorSet annotations = new AnnotationMirrorSet();
```

Comment:
Rather than making this non-final, could `clone()` call `addAll()` on the new `AnnotationMirrorSet` that will be present in the cloned copy? Making things non-final like this seems like a code smell to me.

Reply:
Unfortunately, `super.clone()` makes a shallow copy, so its `AnnotationMirrorSet` is the same object as in the original object.  Calling a mutator like `addAll()` on it will cause a side effect to both the original object and the clone.

I agree that making the variable non-final is gross, but I don't see a better way to do it.